### PR TITLE
feat: [SSCA-625] SLSA Verification step and tests

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

image

image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  CosignSlsaVerifyAttestation,
+  SlsaDockerSourceSpec,
+  SlsaVerificationSource,
+  SlsaVerifyAttestation
+} from 'services/ci'
+import { AllMultiTypeInputTypesForStep } from '../common/utils'
+import { editViewValidateFieldsConfig, transformValuesFieldsConfig } from './SlsaVerificationStepFunctionConfigs'
+import css from '../common/SscaStep.module.scss'
+
+export interface SlsaVerificationStepData {
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepEdit.tsx:41]: Can we not use the same interface defined src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStep.tsx
```
export interface SlsaVerificationStepData {
  name?: string
  identifier: string
  type: string
  timeout?: string
  spec: {
    source: {
      type: SlsaVerificationSource['type']
      spec: SlsaDockerSourceSpec
    }
    verify_attestation: {
      type: SlsaVerifyAttestation['type']
      spec: CosignSlsaVerifyAttestation
    }
  }
}
```

File: src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+): JSX.Element => {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  const isExecutionTimeFieldDisabledForStep = isExecutionTimeFieldDisabled(stepViewType)
+
+  const { getStageFromPipeline, state } = usePipelineContext()
+  const { stage: currentStage } = getStageFromPipeline<BuildStageElementConfig>(
+    state.selectionState.selectedStageId || ''
+  )
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<{
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepEdit.tsx:91]:
```suggestion
  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>
```

File: src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepFunctionConfigs.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    type: ValidationFieldTypes.Timeout,
+    label: 'pipelineSteps.timeoutLabel'
+  }
+]
+
+export const getInputSetViewValidateFieldsConfig = (
+  // Called only with required
+  /* istanbul ignore next */
+  isRequired = true
+): Array<{ name: string; type: ValidationFieldTypes; label?: string; isRequired?: boolean }> => {
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepFunctionConfigs.tsx:102]:
```suggestion
): Array<{ name: string; type: ValidationFieldTypes; label?: StringKeys; isRequired?: boolean }> => {
```

File: src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepInputSet.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          className={cx(stepCss.formGroup, stepCss.sm)}
+        />
+      )}
+
+      {isValueRuntimeInput(get(template, 'spec.verify_attestation.spec.public_key', '')) && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <MultiTypeSecretInput
+            type="SecretFile"
+            expressions={expressions}
+            name={`${prefix}spec.verifyAttestation.publicKey`}
```
Comment (pankajsingh-harness) [inline src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepInputSet.tsx:55]: Should this name not be the same as the key we test for the runtime field?
```suggestion
            name={`${prefix}spec.verify_attestation.spec.public_key`}
```

Comment (udayvunnam) [reply src/modules/75-ssca/components/PipelineSteps/SlsaVerificationStep/SlsaVerificationStepInputSet.tsx:55]: Thank you :)

**Review by deepharness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**